### ISSUE TITLE
Incompatible with Ruby >= 2.6 with recent Webrick version

### ISSUE THREAD

**Issue by nbulaj** [bbtfr/evil-proxy#10]
Hi! There is an issue with Ruby >= 2.6 with a recent Webrick version. Webrick public API was changed and now it requires more arguments.

Check the `do_PUT` / `do_DELETE` / `do_PATCH` / etc methods that calls `perform_proxy_request(req, res)` under the hood. But in a new Webrick it requires 3..4 arguments to be passed:

https://github.com/ruby/webrick/blob/master/lib/webrick/httpproxy.rb#L298

So we catch `wrong number of arguments (given 2, expected 3..4)` exception.

**Comment by drbig** [comment]
Indeed. Perhaps easiest is to RVM an older version... not that this approach has been proven sane (looking at some Pythons...).
